src/libsyntax/codemap.rs

```diff
@@ -592,16 +592,21 @@ impl CodeMap {
         CodeMap {
             files: RefCell::new(Vec::new()),
             expansions: RefCell::new(Vec::new()),
-            file_loader: file_loader
+            file_loader: RefCell::new(file_loader)
         }
     }
 
+    pub fn set_file_loader(&self, file_loader: Box<FileLoader>) {
+        let mut loader = self.file_loader.borrow_mut();
+        *loader = file_loader;
+    }
+
```

Comment:
Mainly because it seemed to be an issue trying to plumb down the file loader through the session in the compiler API, including modifying the entry point. The remaining fields for the codemap follow similar access, so it didn't seem too extreme.